File: c#/leetcode/383-ransom-note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        int[] letters = new int[26];
        
        foreach(char m in magazine) {
            letters[m-'a']++;
        }
                
        foreach(char r in ransomNote) {
            letters[r-'a']--;
        }
        
        return !letters.Any(x => x< 0);
    }
}

File: python/leetcode/383-ransom-note.py
import collections
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mCount = collections.Counter(magazine)
        for ch in ransomNote:
            mCount[ch] = mCount.get(ch, 0) -1
            if mCount[ch] < 0:
                return False
        return True